### ISSUE TITLE
specify deduplication algorithm for custom test types

### ISSUE THREAD

**Issue by fuzzah** [DefectDojo/django-DefectDojo#9000]
### Discussed in https://github.com/DefectDojo/django-DefectDojo/discussions/8972

<div type='discussions-op-text'>

<sup>Originally posted by **fuzzah** November 10, 2023</sup>
Hello!
I import findings via the findings API endpoint with good values in the "unique id from tool" field.
DD seems to ignore this field as it marks my duplicates as just _similar_ findings.
In my case actual duplicates have the same unique id from tool, but almost always their description differs slightly from each other. Title and other fields are the same.
I'd like to configure deduplication algorithm for API-imported findings for my custom test type so that DD would either ignore the description field or rely solely on the unique id from tool field.</div>

So there's this check `if key in DEDUPLICATION_ALGORITHM_PER_PARSER:` in the file `dojo/settings/settings.dist.py`:
```python
# Override the hardcoded settings here via the env var
if len(env('DD_DEDUPLICATION_ALGORITHM_PER_PARSER')) > 0:
    env_dedup_algorithm_per_parser = json.loads(env('DD_DEDUPLICATION_ALGORITHM_PER_PARSER'))
    for key, value in env_dedup_algorithm_per_parser.items():
        if key in DEDUPLICATION_ALGORITHM_PER_PARSER:
            print("Replacing {} with value {} from env var DD_DEDUPLICATION_ALGORITHM_PER_PARSER".format(key, value))
            DEDUPLICATION_ALGORITHM_PER_PARSER[key] = value
```
**Is this check necessary?**
It ignores custom test types as they're not defined in the dict.
I propose to remove the check, as doing so allows user to deduplicate findings in their custom test type with the algorithm they specify in the env var `DD_DEDUPLICATION_ALGORITHM_PER_PARSER`, as I described in the linked discussion.

**Comment by kiblik** [comment]
I suppose the author wanted to avoid to put "some garbage" to `DEDUPLICATION_ALGORITHM_PER_PARSER` and `HASHCODE_FIELDS_PER_SCANNER`.
However, you are right, in my opinion. It is not required to put fields/algo to `DEDUPLICATION_ALGORITHM_PER_PARSER` and `HASHCODE_FIELDS_PER_SCANNER` during the writing parser so the mentioned `if` should be removed from both of these processings.
The community will be happy for any PR which will improve this project.

**Comment by kiblik** [comment]
The issue was fixed in ##9038. Issue can be closed.
